### PR TITLE
Adding Additional Resources to DevTools Pages

### LINKED ISSUE #8165: Adding Additional Resources for DevTools
### Page URL

https://docs.flutter.dev/development/tools/devtools/

### Page source

_No response_

### Describe the problem

I was a part of the Google's Road To GDE Class 4 where I wrote a series of 8 in-depth articles on DevTools under the guidance of Kamal Shree (Flutter GDE).

These articles are very detailed and provide step-by-step codelabs so that people can learn how to effectively use DevTools via real-world scenarios and examples (like Wonderous app, Mandelbrot, etc.). 

Link to all the articles:

- Series Introduction & Installation - [Link](https://medium.com/@fluttergems/mastering-dart-flutter-devtools-series-introduction-installation-part-1-of-8-4f703a8cfcc8)
- Flutter Inspector - [Link](https://medium.com/@fluttergems/mastering-dart-flutter-devtools-flutter-inspector-part-2-of-8-bbff40692fc7)
- App Size Tool - [Link](https://medium.com/@fluttergems/mastering-dart-flutter-devtools-app-size-tool-part-3-of-8-9be6e9ec42a2)
- Network View - [Link](https://medium.com/@fluttergems/mastering-dart-flutter-devtools-network-view-part-4-of-8-afce2463687c)
- Logging View - [Link](https://medium.com/@fluttergems/mastering-dart-flutter-devtools-logging-view-part-5-of-8-b634f3a3af26)
- CPU Profiler View - [Link](https://medium.com/@fluttergems/mastering-dart-flutter-devtools-cpu-profiler-view-part-6-of-8-31e24eae6bf8)
- Memory View - [Link](https://medium.com/@fluttergems/mastering-dart-flutter-devtools-memory-view-part-7-of-8-e7f5aaf07e15)
- Performance View - [Link](https://medium.com/@fluttergems/mastering-dart-flutter-devtools-performance-view-part-8-of-8-4ae762f91230)

Is it possible to add the links to these articles (or selected articles) to the relevant pages of the DevTools section as it would greatly help new Flutter developers who are getting started with DevTools. 

So far, the entire article series has been very well received and appreciated by the Flutter community ([Ref](https://twitter.com/timsneath/status/1608237932388962306?s=20&t=7xnQvYTUBEmFU6xJ_XcaFA)) and adding these articles can definitely help developers in regards to DevTools.

### Expected fix

Adding Aritcle Links to **Codelabs & Other resources** section.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ashitaprasad**
Adding links of the step-by-step tutorials developed under the Google's Road to GDE program to the relevant pages of the DevTools section as it would greatly help new Flutter developers who are getting started with DevTools.

Closes #8165

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/development/tools/devtools/logging.md [hunk 1]
```diff
@@ -38,3 +38,12 @@ click the **Clear logs** button.
...
 [Debugging Flutter apps programmatically]: {{site.url}}/testing/code-debugging
+
+## Other resources
+
+Check out this step-by-step guided [Logging View Tutorial][logging-tutorial] 
+to learn about the various methods of logging available in 
+Flutter and how you can effectively use the Logging View to 
+analyze the logs and debug Flutter apps faster.
+
+[logging-tutorial]: https://medium.com/@fluttergems/mastering-dart-flutter-devtools-logging-view-part-5-of-8-b634f3a3af26
```
Comment (sfshaza2) [inline src/development/tools/devtools/logging.md:49]:
```suggestion
To learn about different methods of logging
and how to effectively use DevTools to
analyze and debug Flutter apps faster,
check out a guided [Logging View tutorial][logging-tutorial].

[logging-tutorial]: {{site.medium}}/@fluttergems/mastering-dart-flutter-devtools-logging-view-part-5-of-8-b634f3a3af26
```

File: src/development/tools/devtools/memory.md [hunk 2]
```diff
@@ -434,3 +434,11 @@ to show its callees.
...
 The bottom-up view shows the list of different
 call stacks that have allocated the instances.
+
+## Other resources
+
+Check out this step-by-step guided [Memory View Tutorial][memory-tutorial] 
+to learn how you can monitor a Flutter app's memory usage 
+and detect any memory leaks using the Memory View.
+
+[memory-tutorial]: https://medium.com/@fluttergems/mastering-dart-flutter-devtools-memory-view-part-7-of-8-e7f5aaf07e15
```
Comment (sfshaza2) [inline src/development/tools/devtools/memory.md:444]:
```suggestion
To learn how to monitor an app's memory usage
and detect memory leaks using DevTools,
check out a guided [Memory View tutorial][memory-tutorial].

[memory-tutorial]: {{site.medium}}/@fluttergems/mastering-dart-flutter-devtools-memory-view-part-7-of-8-e7f5aaf07e15
```

Again, can you confirm whether these articles are behind a paywall?

File: src/development/tools/devtools/network.md [hunk 3]
```diff
@@ -60,4 +60,11 @@ asynchronous timeline events. Viewing network activity in the timeline can be
...
 in your app or in the Flutter framework.
 
+Check out this step-by-step guided [Network View Tutorial][network-tutorial] 
+to learn how to monitor an app's network traffic and inspect 
+different types of requests using the Network View. 
+We eventually use it to identify the network activity that 
+is the cause of the poor app performance.
+
 [timeline]: {{site.url}}/development/tools/devtools/performance#timeline-events-chart
+[network-tutorial]: https://medium.com/@fluttergems/mastering-dart-flutter-devtools-network-view-part-4-of-8-afce2463687c
```
Comment (sfshaza2) [inline src/development/tools/devtools/network.md:70]:
```suggestion
To learn how to monitor an app's network traffic and inspect
different types of requests using the DevTools,
check out a guided [Network View tutorial][network-tutorial].
The tutorial also uses the view to identify network activity that
causes poor app performance.

[timeline]: {{site.url}}/development/tools/devtools/performance#timeline-events-chart
[network-tutorial]: {{site.medium}}/@fluttergems/mastering-dart-flutter-devtools-network-view-part-4-of-8-afce2463687c
```

File: src/development/tools/devtools/performance.md [hunk 4]
```diff
@@ -220,8 +220,16 @@ performance page. To import a performance snapshot, you can drag and drop the
 snapshot into DevTools from any page. **Note that DevTools only
 supports importing files that were originally exported from DevTools.**
 
+## Other resources
+
+Check out the step-by-step guided 
+[Performance View Tutorial][performance-tutorial] to learn 
+how you can monitor an app's perceived performance and 
+detect janks using the Performance View.
+
```
Comment (sfshaza2) [inline src/development/tools/devtools/performance.md:229]:
```suggestion
To learn how to monitor an app's performance and
detect jank using DevTools, check out a guided
[Performance View tutorial][performance-tutorial].
```

File: src/development/tools/devtools/performance.md [hunk 5]
```diff
@@ -220,8 +220,16 @@ performance page. To import a performance snapshot, you can drag and drop the
...
 [generate timeline events]: {{site.developers}}/web/tools/chrome-devtools/evaluate-performance/performance-reference
 [GPU graph]: {{site.url}}/perf/ui-performance#identifying-problems-in-the-gpu-graph
 [Flutter performance profiling]: {{site.url}}/perf/ui-performance
 [Reduce shader compilation jank on mobile]: {{site.url}}/perf/shader
 [Import and export]: #import-and-export
+[performance-tutorial]: https://medium.com/@fluttergems/mastering-dart-flutter-devtools-performance-view-part-8-of-8-4ae762f91230
```
Comment (sfshaza2) [inline src/development/tools/devtools/performance.md:235]:
```suggestion
[performance-tutorial]: {{site.medium}}/@fluttergems/mastering-dart-flutter-devtools-performance-view-part-8-of-8-4ae762f91230
```

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
Thanks for this PR, @ashitaprasad! I've made edits to bring this PR up to Google Developer docs style guidelines. Also, can you confirm whether these articles are behind the Medium paywall?

**Comment by ashitaprasad** [comment]
Thank you for reviewing and editing the PR @sfshaza2.
None of the articles are behind the Medium paywall and do not require any subscription to read.

File: src/development/tools/devtools/app-size.md [hunk 6]
```diff
@@ -259,9 +259,17 @@ In this example, import the `build/apk-code-size-analysis_01.json`
...
+the Wonderous App using the App Size Tool. Various strategies 
+are also discussed that can help reduce an app's size.
+
 [Using the treemap]: #using-the-treemap
 [Generating size files]: #generating-size-files
 [Analysis tab]: #analysis-tab
 [Diff tab]: #diff-tab
 [installation instructions]: {{site.url}}/development/tools/devtools/overview#install-devtools
 [App Size Documentation]: {{site.url}}/perf/app-size#breaking-down-the-size
+[app-size-tutorial]: https://medium.com/@fluttergems/mastering-dart-flutter-devtools-app-size-tool-part-3-of-8-9be6e9ec42a2
```
[the remark this answers is not in the document]
Comment (ashitaprasad) [reply src/development/tools/devtools/app-size.md]: These articles are not behind any paywall.

File: src/development/tools/devtools/memory.md (hunk 2 above)
Comment (ashitaprasad) [reply src/development/tools/devtools/memory.md:444]: Yes, none of these articles are behind the paywall.

**Comment by sfshaza2** [comment]
That's so great to hear, @ashitaprasad! Wonderful news!

File: src/development/tools/devtools/performance.md [hunk 7]
```diff
@@ -220,8 +220,14 @@ performance page. To import a performance snapshot, you can drag and drop the
 snapshot into DevTools from any page. **Note that DevTools only
 supports importing files that were originally exported from DevTools.**
 
+## Other resources
+
+To learn how to monitor an app's performance and
+detect jank using DevTools, check out a guided
+[Performance View tutorial][performance-tutorial].
```
Comment (parlough) [inline src/development/tools/devtools/performance.md:227]: Really cool articles!

I think the build is failing due to these Markdown link references not resolving. An extra line should fix that:

```suggestion
[Performance View tutorial][performance-tutorial].

```

Comment (ashitaprasad) [reply src/development/tools/devtools/performance.md:227]: Thank a lot @parlough 🙂
Also, thank you for suggesting the fix. I was wondering why the link check failed.

**Comment by sfshaza2** [comment]
Yes, thanks, @parlough! And thanks, @ashitaprasad!

**Review by sfshaza2 — APPROVED** [review]
lgtm

**Pull request merged**